src/js/components/interactiveDataSources/lottieAnimation/LottieAnimation.jsx

```diff
@@ -0,0 +1,227 @@
+ * `onStepEnter` callback.
+ *
+ * In the below example, the animation JSON is used as the backdrop for the
+ * `ScrollerOverlay` component, which contains a card with text. When the first
+ * `ScrollerOverlay` component scrolls into the offset threshold, `animation-1` will
+ * play from `0` frames to `500` frames. When the second `ScrollerOverlay`
+ * component scrolls into the offset threshold, `animation-2` will play from `200` to `700`
+ * frames. Add the `isScrollerBackdrop` prop to have the scroll direction determine the play
+ * direction: forward when scrolling down into the threshold and in reverse when
+ * scrolling back up into the threshold.
+ *
+        function ExampleScroller() {
+            const lottieRef1 = useRef();
+            const lottieRef2 = useRef();
+            return (
+                <div>
+                    <Scroller>
+                        <LottieAnimation
+                            isScrollerBackdrop
+                            name="animation-1"
+                            ref={lottieRef1}
+                            src="/img/animations/animation1.json" />
+                        <LottieAnimation
+                            isScrollerBackdrop
+                            name="animation-2"
+                            ref={lottieRef2}
+                            src="/img/animations/animation2.json" />
+
+                        <ScrollerOverlay
+                            content="animation-1"
+                            onStepEnter={() =>
+                                lottieRef1.current?.playAnimation(0, 500)
+                            }>
+                                <p className="scroller-card">
+                                    Overlay card for animation 1
+                                </p>
+                        </ScrollerOverlay>
+                        <ScrollerOverlay
+                            content="animation-2"
+                            onStepEnter={() =>
+                                lottieRef2.current?.playAnimation(200, 700)
+                            }>
+                                <p className="scroller-card">
+                                    Overlay card for animation 2
+                                </p>
+                        </ScrollerOverlay>
+                    </Scroller>
+                </div>
+            )
+        }
+        export default ExampleScroller;
+ *
+ */
+class LottieAnimation extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            src: this.props.src || null, // string path to lottie animation json
+            autoplay: this.props.autoplay || false,
+            loop: this.props.loop || false,
+            fps: this.props.fps || 60, // frames per second
+            isScrollerBackdrop: this.props.isScrollerBackdrop || false, // if used as `Scroller` backdrop
+            direction: 'down',
+            oldScrollY: 0
```

Comment:
Here is where the state for various things are being initialized with any props that are passed down. If no props are passed down to initialize the state, then the default/fallback value after the OR ( || ) symbol is used.

Reply:
Sorry I specifically meant `oldScrollY`

Reply:
That's used to determine on whether or not the scrolling has receded or advanced to determine whether or not to play the animation in reverse. It is used in the function, `this.setDirectionState` where it is compared against the newer/current scrollY value. I initialized it to 0 since that's the top of the page.

Reply:
☝️ would be a great thing to add as a comment in the code